macros/models/_sector/nft/enrich_trades.sql

```diff
@@ -0,0 +1,107 @@
+{% macro enrich_trades(blockchain='', models=[], transactions_model=null, tokens_nft_model=null, tokens_erc20_model=null, prices_model=null, aggregators=null, aggregator_markers=null) %}
+-- Macro to apply the NFT trades enrichment(s) to base models
+-- 1. add transaction information
+-- 2. add NFT token information
+-- 3. add ERC20 token information + handle ERC20 decimals
+-- 4. handle USD columns
+-- 5. add aggregator columns
+-- 6. fix buyer or seller for aggregator txs
+-- 7. calculate platform and royalty rates
+
+WITH base_union as (
+    {% for nft_model in models %}
+    SELECT
+        '{{ blockchain }}' as blockchain,
+        '{{ nft_model[0] }}' as project,
+        '{{ nft_model[1] }}' as project_version,
+        block_date,
+        block_number,
+        tx_hash,
+        project_contract_address,
+        trade_category,                 --buy/sell/swap
+        trade_type,                     --primary/secondary
+        buyer,
+        seller,
+        nft_contract_address,
+        nft_token_id,
+        nft_amount,                -- always 1 for erc721
+        price_raw,
+        currency_contract,
+        platform_fee_amount_raw,
+        royalty_fee_amount_raw,
+        platform_fee_address,   -- optional
+        royalty_fee_address,    -- optional
+        sub_tx_trade_id
+    FROM {{ nft_model[2] }}
+    {% if not loop.last %}
+    UNION ALL
+    {% endif %}
+    {% endfor %}
+),
+
+enrichments as (
+SELECT
+    base.blockchain,
+    base.project,
+    base.project_version,
+    base.block_date,
+    base.block_number,
+    base.tx_hash,
+    base.sub_tx_trade_id,
+    base.project_contract_address,
+    base.trade_category,
+    base.trade_type,
+    case when base.buyer != agg.contract_address then base.buyer else tx.from end as buyer,
+    case when base.seller != agg.contract_address then base.seller else tx.from end as seller,
+    base.nft_contract_address,
+    base.nft_token_id,
+    base.nft_amount,
+    base.price_raw,
+    base.currency_contract,
+    base.platform_fee_amount_raw,
+    base.royalty_fee_amount_raw,
+    base.platform_fee_address,
+    base.royalty_fee_address,
+    tx.block_time,
```

Comment:
any reason we use `block_time` from base transactions source? rather than decoded source(s)?

Reply:
It's not part of the base schema currently.
But now that I think of it, it might be better included there..